src/ncp/rcp_host.cpp

```diff
@@ -331,19 +331,28 @@ void RcpHost::Deinit(void)
     mThreadEnabledStateChangedCallbacks.clear();
     mResetHandlers.clear();
 
+    mJoinReceiver              = nullptr;
     mSetThreadEnabledReceiver  = nullptr;
     mScheduleMigrationReceiver = nullptr;
     mDetachGracefullyCallbacks.clear();
 }
 
 void RcpHost::HandleStateChanged(otChangedFlags aFlags)
 {
+    otOperationalDatasetTlvs datasetTlvs;
+    GetDatasetActiveTlvs(datasetTlvs);
```

Comment:
I wonder if we need to do this every time there is a state change. Maybe only when there is `OT_CHANGED_ACTIVE_DATASET` change?

Reply:
Good idea! I move the checking of dataset into the if loop. So it will only be fetched when other conditions are all met.